platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/TestEventReporterFactory.java

```diff
@@ -36,8 +38,16 @@ public interface TestEventReporterFactory {
      * </p>
      *
      * @param rootName the name for the root node of the test tree
+     * @param binaryResultsDirectory the directory to write binary test results to
+     * @param htmlReportDirectory the directory to write HTML test reports to
+     *
      * @return the test event reporter
+     *
      * @since 8.12
      */
-    GroupTestEventReporter createTestEventReporter(String rootName);
+    GroupTestEventReporter createTestEventReporter(
+        String rootName,
+        Path binaryResultsDirectory,
+        Path htmlReportDirectory
```

Comment:
What about `Directory`?

Reply:
Sure can change that.

I personally have a dislike with these types. I'm not sure why Gradle decided to reinvent a whole new file API. 

After a conversation with Lorant in the past, we sorta came to the conclusion that Gradle shouldn't try to make its own file API and should rely on the stdlib.

For example, if we want to enforce that a file is a Directory, we _still_ have `@InputFiles`, `@InputDirectory`, etc.

I don't actually see `Directory` helping us here.